evalml/automl/automl_search.py

```diff
@@ -466,11 +467,11 @@ def __init__(
                     self.sampler_method,
                     self.sampler_balanced_ratio,
                 )
-            if self._sampler_name not in parameters:
+            if self._sampler_name not in parameters and self._sampler_name is not None:
```

Comment:
We use this to make sure we don't add the sampling ratio to the param dictionary if the `sampler_name` is None

Reply:
Was this a bug?

Reply:
@freddyaboulton I think so. It wasn't doing anything except adding `{None: {sampler_ratio: 0.25}}` to the param dictionary, which doesn't do anything but also isn't necessary and could cause problems in the future (like with this PR). It should be fixed now!